### ISSUE TITLE
script not work on zabbix server (Source compile version)

### ISSUE THREAD

**Issue by naechin91004** [ktelep/EMC-Zabbix-Integration#9]
Hello.

I was changed zabbix server package to zabbix server source on same server.
I just delete package version and I install source version.
No change another.

After script not work with below error.

  File "/usr/lib/zabbix/externalscripts/emc_vnx_stats.py", line 185
    print "------------------------------------------------------"
                                                                 ^
SyntaxError: Missing parentheses in call to 'print'

And I install below package for compile
yum install gcc gcc-c++ make openssl-devel curl wget net-snmp net-snmp-utils net-snmp-libs net-snmp-devel gnutls gnutls-devel libxml2 libxml2-devel curl-devel
yum install httpd php php-mysql php-gd pgp-cli php-xml php-bcmath php-mbstring mod_ssl openssl

And I complied below command.
./configure --enable-server --enable-agent --with-mysql --enable-ipv6 --with-net-snmp --with-libcurl --with-libxml2
make
make install

**Comment by ktelep** [comment]
Can you confirm what version of python is installed?   You would need python 2.7, it's possible that it was removed when you uninstalled the packaged version?

The error would indicate it's trying to execute it with python 3.0

**Comment by naechin91004** [comment]
I made new server for test.
that is centos 7 & python 2.7.5. and Error change look like below

Traceback (most recent call last):
  File "./emc_vnx_discovery", line 480, in <module>
    main()
  File "./emc_vnx_discovery", line 461, in main
    result = discover_array_disks(ecom_conn, args.serial)
  File "./emc_vnx_discovery", line 88, in discover_array_disks
    array = get_array_instancename(array_serial, ecom_conn)
  File "./emc_vnx_discovery", line 29, in get_array_instancename
    registered_arrays = ecom_conn.EnumerateInstanceNames("Clar_StorageSystem")
  File "/usr/lib/python2.7/site-packages/pywbem/cim_operations.py", line 398, in EnumerateInstanceNames
    **params)
  File "/usr/lib/python2.7/site-packages/pywbem/cim_operations.py", line 187, in imethodcall
    raise CIMError(0, str(arg))
pywbem.cim_operations.CIMError: (0, 'SSL error: certificate verify failed')

**Comment by ktelep** [comment]
There is a known issue with certain versions of pywbem and CentOS. I need
to determine a good way to get around this.

The quick way to get running is to change the https in the ECOM connect
string to http but that's not a good end state option. Let me dig into this
and see if I can build in a better workaround.

Can you provide me with what version of pywbem is installed?

**Comment by naechin91004** [comment]
I install pywbem-0.7.0-25.20130827svn625.el7.noarch version

**Comment by naechin91004** [comment]
I add This one and script work

```
## add  no_verification=True,
return pywbem.WBEMConnection(ecom_url, (ecom_user, ecom_pass), no_verification=True,
                             default_namespace="/root/emc")
```

**Comment by zielony360** [comment]
@naechin91004 Thanks! I also had to add this.

**Comment by ktelep** [comment]
no_verification will bypass the SSL verification, so is a valid workaround, not the best from a security perspective however.   The issue has apparently been resolved in pywbem 0.8, so possibly upgrading to this version will resolve the issue without having to set no_verification to True.

**Comment by ktelep** [comment]
Closed as there is the workaround listed above and being resolve din pywbem 0.8
